File: OnlineGroceryOrderingSystemAuthenticationJava/src/main/java/com/alpha/ogo/service/UserService.java
package com.alpha.ogo.service;

import com.alpha.ogo.domain.Address;
import com.alpha.ogo.domain.User;
import com.alpha.ogo.dto.CommonResponce;

/**
 * @author Tharindu
 *
 * @Date Jun 18, 2021
 *
 */
public interface UserService {

	CommonResponce<User> createUser(User user);
	CommonResponce<User> checkUsername(String username);
	CommonResponce<User> getAllUsers();
	CommonResponce<User> getUserById(Long userId);
	
	CommonResponce<Address> createAddress(Address address);
	CommonResponce<Address> getAllUserAddress(Long userId);

}

File: OnlineGroceryOrderingSystemAuthentication/src/main/kotlin/com/allion/ogoss/service/UserService.kt
package com.allion.ogoss.service

import com.allion.ogoss.domain.Address
import com.allion.ogoss.domain.Registration
import com.allion.ogoss.domain.User
import com.allion.ogoss.dto.CommonResponce

interface UserService {

    fun createUserRegistration(user: User?): CommonResponce<User>

    fun createUserAddress(address: Address?): CommonResponce<Address>

    fun getAllUserAddress(userId :Long?): CommonResponce<Address>

    fun getAddressById(addressId: Long?): CommonResponce<Address>

    fun getUserDetailById(userId: Long?): CommonResponce<User>


    fun createRegistration(registration: Registration?): CommonResponce<Registration>

    fun checkUsername(username: String?): CommonResponce<User?>?

    fun getAllUsers(): CommonResponce<Registration?>?

    fun getRegistrationById(registrationId: String?): CommonResponce<Registration?>?



}
